src/mocks/mock-request.js

```diff
@@ -2,22 +2,29 @@ import MockAssert from './mock-assert'
 import Response from '../response'
 import { isPlainObject } from '../utils/index'
 import { clone } from '../utils/clone'
-import { sortedUrl, toSortedQueryString, isSubset } from './mock-utils'
+import { sortedUrl, toSortedQueryString, filterKeys } from './mock-utils'
 
 /**
  * @param {number} id
  * @param {object} props
  *   @param {string} props.method
  *   @param {string|function} props.url
  *   @param {string|function} props.body - request body
+ *   @param {string} props.mockName
  *   @param {object} props.response
  *     @param {string} props.response.body
  *     @param {object} props.response.headers
  *     @param {integer} props.response.status
  */
+
+const MATCHED_AS_UNDEFINED_IN_MOCK = 'MATCHED_AS_UNDEFINED_IN_MOCK'
+const MATCHED_BY_FUNCTION = 'MATCHED_BY_FUNCTION'
+const MISMATCH_BY_FUNCTION = 'MISMATCHED_BY_FUNCTION'
+
 function MockRequest(id, props) {
   this.id = id
 
+  this.mockName = props.mockName ? props.mockName : this.id
```

Comment:
Maybe interpolate a name from the id, something like:

```suggestion
  this.mockName = props.mockName ? props.mockName : `Mock#${this.id}`
```